### PR TITLE
SERVER-1716

### PR DESCRIPTION
**Pull request by can-keklik**

### CODE REVIEW

**Comment by ar2rsawseen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- api/utils/rights.js  15
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5180117&bid=17000181)

**Comment by ar2rsawseen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- api/utils/rights.js  15
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5180117&bid=17000181)

File: api/utils/rights.js [hunk 1]
```diff
@@ -391,4 +394,161 @@ function wrapCallback(params, callback, callbackParam, func) {
...
+        }
+    }
+    var appList = [];
+    if (collection.indexOf("events") === 0 || collection.indexOf("drill_events") === 0) {
+        for (let i = 0; i < apps.length; i++) {
+            if (apps[i].length) {
+                appList.push({_id: apps[i]});
+            }
+        }
+        dbLoadEventsData(params, appList, function(err, eventList/*, viewList*/) {
```
Comment (ronkorving) [inline api/utils/rights.js]: Missing error handling

File: api/utils/rights.js [hunk 2]
```diff
@@ -391,4 +394,161 @@ function wrapCallback(params, callback, callbackParam, func) {
...
+        });
+    }
+    else if (collection.indexOf("app_viewdata") === 0) {
+        for (let i = 0; i < apps.length; i++) {
+            if (apps[i].length) {
+                appList.push({_id: apps[i]});
+            }
+        }
+
+        dbLoadEventsData(params, appList, function(err, eventList, viewList) {
```
Comment (ronkorving) [inline api/utils/rights.js]: Same concern here

File: api/utils/rights.js (hunk 1 above)
Comment (can-keklik) [reply api/utils/rights.js]: Added handling: https://github.com/Countly/countly-server/pull/1207

File: api/utils/rights.js (hunk 2 above)
Comment (can-keklik) [reply api/utils/rights.js]: Added handling: https://github.com/Countly/countly-server/pull/1207

**Pull request merged**